src/pg_query_summary.c

```diff
@@ -0,0 +1,775 @@
+			return false; // Not handled by tree walker.
+		}
+
+		case T_AlterTableStmt: {
+			AlterTableStmt* stmt = castNode(AlterTableStmt, node);
+
+			// `ALTER INDEX index_name` is ignored.
+			if (stmt->objtype != OBJECT_INDEX)
+				add_from_clause_item((Node*) stmt->relation, CONTEXT_DDL, state);
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_CreateStmt:
+			add_from_clause_item((Node*) castNode(CreateStmt, node)->relation, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+
+		case T_CreateTableAsStmt: {
+			CreateTableAsStmt* stmt = castNode(CreateTableAsStmt, node);
+			if (stmt->into && stmt->into->rel)
+				add_from_clause_item((Node*) stmt->into->rel, CONTEXT_DDL, state);
+
+			pg_query_summary_walk_impl(stmt->query, state);
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_TruncateStmt:
+			add_from_clause_items(castNode(TruncateStmt, node)->relations, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+
+		case T_ViewStmt: {
+			ViewStmt* stmt = castNode(ViewStmt, node);
+			add_from_clause_item((Node*) stmt->view, CONTEXT_DDL, state);
+			pg_query_summary_walk_impl(stmt->query, state);
+			return false; // Not handled by tree walker.
+		}
+
+		case T_IndexStmt: {
+			IndexStmt* stmt = castNode(IndexStmt, node);
+
+			add_from_clause_item((Node*) stmt->relation, CONTEXT_DDL, state);
+
+			ListCell* lc = NULL;
+			foreach(lc, stmt->indexParams) {
+				IndexElem* index_elem = lfirst(lc);
+				pg_query_summary_walk_impl(index_elem->expr, state);
+			}
+
+			pg_query_summary_walk_impl(stmt->whereClause, state);
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_CreateTrigStmt:
+			add_from_clause_item((Node*) castNode(CreateTrigStmt, node)->relation, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+
+		case T_RuleStmt:
+			add_from_clause_item((Node*) castNode(RuleStmt, node)->relation, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+
+		case T_VacuumStmt: {
+			VacuumStmt* stmt = castNode(VacuumStmt, node);
+
+			ListCell* lc = NULL;
+			foreach(lc, stmt->rels) {
+				VacuumRelation* rel = lfirst(lc);
+				add_from_clause_item((Node*) rel->relation, CONTEXT_DDL, state);
+			}
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_RefreshMatViewStmt:
+			add_from_clause_item((Node*) castNode(RefreshMatViewStmt, node)->relation, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+
+		case T_DropStmt: {
+			DropStmt* stmt = castNode(DropStmt, node);
+
+			switch (stmt->removeType) {
+				case OBJECT_TABLE: {
+					ListCell* lc = NULL;
+					foreach(lc, stmt->objects) {
+						Node* obj = lfirst(lc);
+						if (obj->type == T_List) {
+							List* list = castNode(List, obj);
+
+							if (list_length(list) == 0)
+								continue;
+
+							RangeVar* range_var = makeRangeVarFromNameList(list);
+							add_table(state, CONTEXT_DDL, range_var);
```

Comment:
For consistency, we can use (the now renamed) `add_range_var_with_context` here as well:
```suggestion
							add_range_var_with_context(state, CONTEXT_DDL, makeRangeVarFromNameList(list));
```